app/assets/javascripts/actions/revisions_actions.js

```diff
@@ -9,9 +10,25 @@ import {
 import { fetchWikidataLabelsForRevisions } from './wikidata_actions';
 import logErrorMessage from '../utils/log_error_message';
 import request from '../utils/request';
+import { fetchRevisionsFromUsers } from '../utils/mediawiki_revisions_utils';
+import { fetchRevisionsAndReferences } from './media_wiki_revisions_actions';
+import { sortRevisionsByDate } from '../utils/revision_utils';
+
+const fetchRevisionsPromise = async (course, limit, isCourseScoped, users, last_date, lastRevisions, dispatch) => {
```

Comment:
I think this makes the isCourseScoped vs normal requests so different that they shouldn't be handled in the same function. It will be easier to reason about if there is a separate function for course-scoped revision requests. Probably at the component level, it should just call a different action (`fetchRevisions` vs `fetchCourseScopedRevisions`) so that the data flows are totally separate.